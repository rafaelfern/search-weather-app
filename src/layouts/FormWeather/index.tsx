import React, { LegacyRef } from 'react'

export interface IFormWeatherProps {
    handleSubmit: (inputValue: any) => void
    inputRef: LegacyRef<HTMLInputElement> | undefined
}

export default function FormWeather({ handleSubmit, inputRef}: IFormWeatherProps) {
  return (
    <>
    <h1 className="title mb-6 mt-4 py-4 text-5xl font-serif">Search Weather</h1>
        <div className="form">
            <form onSubmit={handleSubmit}>
                <input 
                    ref={inputRef}
                    type="text" 
                    placeholder='Enter city name'
                    className="px-4 py-3 rounded-sm mr-2"
                    onChange={() => {}}
                />
                <button type="submit" className='btn px-4 py-3 text-white rounded-sm'>
                    Search
                </button>
            </form>
        </div>
    </>
  )
}
