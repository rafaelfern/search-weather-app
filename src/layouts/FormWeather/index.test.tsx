import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import FormWeather, { IFormWeatherProps } from './index'

const props: IFormWeatherProps = {
    handleSubmit: () => {},
    inputRef: undefined
}

describe('<Home />', () => {
    it('should render the component an header title', () => {
        const { container } = render(<FormWeather {...props} />);
        const heading = screen.getByRole('heading', {
            name: /search Weather/i
        })

        expect(heading).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
})