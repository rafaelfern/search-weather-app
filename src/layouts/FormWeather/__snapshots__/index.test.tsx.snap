// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render the component an header title 1`] = `
<div>
  <h1
    class="py-4 text-5xl text-white font-serif"
  >
    Search Weather
  </h1>
  <div
    class="form"
  >
    <form>
      <input
        class="px-4 py-3 rounded-sm mr-2"
        placeholder="Enter city name"
        type="text"
      />
      <button
        class="px-4 py-3 bg-purple-500 text-white rounded-sm"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;
